llvm/lib/Transforms/InstCombine/InstCombineCalls.cpp

```diff
@@ -4071,9 +4071,7 @@ bool InstCombinerImpl::transformConstExprCastCall(CallBase &Call) {
       if (Callee->isDeclaration())
         return false;   // Cannot transform this return value.
 
-      if (!Caller->use_empty() &&
-          // void -> non-void is handled specially
-          !NewRetTy->isVoidTy())
+      if (!Caller->use_empty() || NewRetTy->isVoidTy())
```

Comment:
```suggestion
      if (!Caller->use_empty())
```
A more conservative change would be this. In that case we'd still allow changing the return type to void as long as the return value is not used. That may be more consistent with the overall handling?

Reply:
Yes this makes sense.